### ISSUE TITLE
Compilation error when building tfcompile on Windows 10

### ISSUE THREAD

**Issue by GPhilo** [tensorflow/tensorflow#24218]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): r1.12
- Python version: 3.5
- Bazel version (if compiling from source): 0.15
- GCC/Compiler version (if compiling from source): VS 2015 14.0.25431.01 Update 3
- CUDA/cuDNN version: 9.0/7.3.1
- GPU model and memory: GTX 1080, 8GBs

**Describe the current behavior**
Checked out the release 1.12 branch of the TF repo, configured bazel (configuration is below) and ran 
`bazel build -c opt --config=opt //tensorflow/compiler/aot:tfcompile`

Compilation fails at `.\tensorflow/compiler/tf2xla/cpu_function_runtime.h(71): error C2338`, it seems a static assert is failing: 
```
std::pair<uint64, uint64> Encode() const {
    static_assert(sizeof(*this) == 16, ""); //<<<<<<<<<<<<<<<<< 
    uint64 upper = Pack(kind(), size_);
    uint64 lower = entry_param_number_;
    return {upper, lower};
  }
```

**Describe the expected behavior**
The code compiles.

**Other info / logs**

Bazel config:

```
build --action_env PYTHON_BIN_PATH="C:/Users/1/Anaconda3/envs/tensorflow/python.exe"
build --action_env PYTHON_LIB_PATH="C:/Users/1/Anaconda3/envs/tensorflow/lib/site-packages"
build --python_path="C:/Users/1/Anaconda3/envs/tensorflow/python.exe"
build:ignite --define with_ignite_support=true
build:xla --define with_xla_support=true
build --action_env TF_NEED_OPENCL_SYCL="0"
build --action_env TF_NEED_ROCM="0"
build --action_env TF_NEED_CUDA="1"
build --action_env CUDA_TOOLKIT_PATH="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.0"
build --action_env TF_CUDA_VERSION="9.0"
build --action_env CUDNN_INSTALL_PATH="C:/Program Files/NVIDIA GPU Computing Toolkit/cuDNN/7.3.1"
build --action_env TF_CUDNN_VERSION="7"
build --action_env TF_CUDA_COMPUTE_CAPABILITIES="5.2,6.1"
build --action_env TF_CUDA_CLANG="0"
build --config=cuda
test --config=cuda
build:opt --copt=/arch:AVX2
build:opt --define with_default_optimizations=true
build --config monolithic
build --copt=-w --host_copt=-w
build --verbose_failures
build --distinct_host_configuration=false
build --experimental_shortened_obj_file_path=true
build --define=no_tensorflow_py_deps=true
build --define=override_eigen_strong_inline=true
build:v2 --define=tf_api_version=2
```

**Comment by gunan** [comment]
I do not think we have XLA support on windows yet.
Reassigning to Justin to reroute the issue in XLA team.
Justin, is anyone working on XLA on windows?

**Comment by jlebar** [comment]
I don't believe anyone is working on XLA support on Windows, though I would certainly review a PR that fixed this.  I believe this static_assert *is* checking something important.

**Comment by GPhilo** [comment]
As a side question, then: Is the fact that windows is not currently supported written anywhere in the documentation?
I went through the pages of the [XLA documentation](https://www.tensorflow.org/xla/overview) and found no mention of specific OSes or supported configurations.
This is IMO a **very** important information to have there. Had I not posted this question, I would have never known Windows is not supported, and I posted this only after several attempts following multiple guides and with multiple TF versions. That's a lot of time spent on the lack of a "currently we only support builds on Linux" anywhere I could find in the docs. Or, if that's not possible for whatever reason, please at least include a compiler check that gives a *maningful* error message instead of an empty string.

**Comment by farimani** [comment]
Vistual Studio does not pack adjacent bitfields if they are of different type. In this case, "kind" is of an enum type, treated as an unsigned int (32 bits), and "size" is uint64. The resulting class is 8+16 bytes in size. The fix is to declare "kind" also as uint64 and typecast the assignments. But even when you fix that you will run into other issues with files generated by the protocol buffer compiler. I haven't chased those down.

**Comment by jvishnuvardhan** [comment]
@farimani Is this still an issue with new TF version? Could you check whether the issue persists? Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by imnotkind** [comment]
@jvishnuvardhan I'm trying to buikd TensorFlow 1.13.0-rc2, and I still have this issue.

**Comment by jvishnuvardhan** [comment]
@imnotkind Please open a new ticket by filling the issue [template](https://github.com/tensorflow/tensorflow/issues/new?template=10-build-installation-issue.md) and providing as many details as possible. Thanks!

**Comment by GPhilo** [comment]
@jvishnuvardhan what's the point of opening an identical issue to this one just to bump the TF version it refers to? The issue was closed for lack of activity, but the lack is not on the user's side. Is XLA on windows still unsupported? Has the error message been improved to clarify why the static assert fails? (Considering that [the offending line is still unchanged in the master branch](https://github.com/tensorflow/tensorflow/blob/00bd52b61dc7be94c09cbd7be3c3e14629ceee7c/tensorflow/compiler/tf2xla/cpu_function_runtime.h#L71) I have a feeling nothing changed).

**Comment by GPhilo** [comment]
@jvishnuvardhan since I'm also still having the issue with 1.13-rc2 I opened a new issue with the full current configuration.
